bgpd/bgp_vty.c

```diff
@@ -2426,6 +2435,43 @@ DEFUN(no_bgp_minimum_holdtime, no_bgp_minimum_holdtime_cmd,
 	return CMD_SUCCESS;
 }
 
+DEFUN(bgp_tcp_keepalive, bgp_tcp_keepalive_cmd,
+      "bgp tcp-keepalive (1-65535) (1-65535) (1-30)",
+      "BGP specific commands\n"
+      "TCP keepalive parameters\n"
+      "TCP keepalive idle time (seconds)\n"
+      "TCP keepalive interval (seconds)\n"
+      "TCP keepalive maximum probes\n")
+{
+	uint32_t idle, intvl, probes;
+	VTY_DECLVAR_CONTEXT(bgp, bgp);
+
+	idle = strtoul(argv[2]->arg, NULL, 10);
+	intvl = strtoul(argv[3]->arg, NULL, 10);
+	probes = strtoul(argv[4]->arg, NULL, 10);
+
+	bgp_tcp_keepalive_set(bgp, (uint16_t)idle, (uint16_t)intvl,
+			      (uint16_t)probes);
+
+	return CMD_SUCCESS;
+}
+
+DEFUN(no_bgp_tcp_keepalive, no_bgp_tcp_keepalive_cmd,
+      "no bgp tcp-keepalive (1-65535) (1-65535) (1-30)",
```

Comment:
Can numbers be optional for negative form?

Reply:
ok